components/tidb_query_expr/src/impl_math.rs

```diff
@@ -493,6 +530,28 @@ fn truncate_real(x: Real, d: i32) -> Real {
     }
 }
 
+#[inline]
+#[rpn_fn]
+pub fn truncate_decimal_with_int(arg0: &Decimal, arg1: &Int) -> Result<Option<Decimal>> {
+    let d = if *arg1 >= 0 {
+        *arg1.min(&127) as i8
+    } else {
+        *arg1.max(&-128) as i8
+    };
+
+    let res: codec::Result<Decimal> = arg0.to_owned().round(d, RoundMode::Truncate).into();
```

Comment:
- Do we need to use `to_owned()` here? `Decimal` is ok to `Copy`.
- Can we use `into()?` directly here?

Reply:
ping @wangggong to take a look.

Reply:
1. Yes, `Decimal` is ok to `Copy`. I removed  it.
2. If the code just like

```rust
let res: codec::Result<Decimal> = arg0.into();
```

The error will be raised:

```
^^^^ the trait `std::convert::From<&tidb_query_datatype::codec::data_type::Decimal>` is not implemented for `std::result::Result<tidb_query_datatype::codec:
 :data_type::Decimal, tidb_query_datatype::codec::Error>`
```

Thus here `round(d, RoundMode::Truncate)` is not removed now.

Reply:
What I mean is:

```
let res = arg0.round(d, RoundMode::Truncate).into()?; 
```

I don't know whether or not this can be compiled.

Reply:
```rust
let res = arg0.round(d, RoundMode::Truncate).into()?;  // This will raise error: error[E0283]: type annotations needed
// let res: Decimal = arg0.round(d, RoundMode::Truncate).into()?;  // This will also raise the error, cause `into` cannot infer type.
```

Thus here the code will not be modified.

Reply:
OK, I got it.